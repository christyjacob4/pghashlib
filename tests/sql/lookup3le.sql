-- Test basic hash functionality with text
SELECT lookup3le('hello world');

-- Test hash with different text inputs
SELECT lookup3le('test string');
SELECT lookup3le('another test');

-- Test text input with custom initval
SELECT lookup3le('hello world', 42);
SELECT lookup3le('hello world', 123);

-- Test bytea input
SELECT lookup3le('hello world'::bytea);

-- Test bytea input with custom initval
SELECT lookup3le('hello world'::bytea, 42);

-- Test integer input
SELECT lookup3le(12345);
SELECT lookup3le(-12345);

-- Test integer input with custom initval
SELECT lookup3le(12345, 42);
SELECT lookup3le(-12345, 123);

-- Test consistency (same input should give same hash)
SELECT lookup3le('consistent test') = lookup3le('consistent test');

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup3le('initval test', 1) != lookup3le('initval test', 2);

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup3le'
ORDER BY proname, proargtypes;

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';