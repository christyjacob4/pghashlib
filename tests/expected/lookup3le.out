-- Test basic hash functionality with text
SELECT lookup3le('hello world');
 lookup3le  
------------
 1252609637
(1 row)

-- Test hash with different text inputs
SELECT lookup3le('test string');
 lookup3le 
-----------
 546331042
(1 row)

SELECT lookup3le('another test');
  lookup3le  
-------------
 -2135089103
(1 row)

-- Test text input with custom initval
SELECT lookup3le('hello world', 42);
 lookup3le  
------------
 -419950101
(1 row)

SELECT lookup3le('hello world', 123);
 lookup3le  
------------
 -656773485
(1 row)

-- Test bytea input
SELECT lookup3le('hello world'::bytea);
 lookup3le  
------------
 1252609637
(1 row)

-- Test bytea input with custom initval
SELECT lookup3le('hello world'::bytea, 42);
 lookup3le  
------------
 -419950101
(1 row)

-- Test integer input
SELECT lookup3le(12345);
 lookup3le  
------------
 1490861095
(1 row)

SELECT lookup3le(-12345);
 lookup3le  
------------
 -983645695
(1 row)

-- Test integer input with custom initval
SELECT lookup3le(12345, 42);
 lookup3le  
------------
 -483149855
(1 row)

SELECT lookup3le(-12345, 123);
  lookup3le  
-------------
 -1537810365
(1 row)

-- Test consistency (same input should give same hash)
SELECT lookup3le('consistent test') = lookup3le('consistent test');
 ?column? 
----------
 t
(1 row)

-- Test initval effect (same input, different initvals should give different hashes)
SELECT lookup3le('initval test', 1) != lookup3le('initval test', 2);
 ?column? 
----------
 t
(1 row)

-- Test function properties
SELECT 
    proname,
    provolatile,
    proisstrict
FROM pg_proc 
WHERE proname = 'lookup3le'
ORDER BY proname, proargtypes;
  proname  | provolatile | proisstrict 
-----------+-------------+-------------
 lookup3le | i           | t
 lookup3le | i           | t
 lookup3le | i           | t
 lookup3le | i           | t
 lookup3le | i           | t
 lookup3le | i           | t
(6 rows)

-- Test extension metadata
SELECT 
    extname,
    extversion
FROM pg_extension 
WHERE extname = 'hashlib';
 extname | extversion 
---------+------------
 hashlib | 0.0.1
(1 row)

